src/event.processor.ts

```diff
@@ -0,0 +1,142 @@
+import axios from "axios";
+
+export class EventProcessor {
+  private options: EventProcessorOptions = new EventProcessorOptions();
+
+  async start(options: EventProcessorOptions) {
+    this.options = options;
+
+    if (!options.emitterAddresses && !options.eventIdentifiers) {
+      throw new Error(`No emitter address or identifier set. Could not resolve all the events without filters.`);
+    }
+
+    await this.startProcess(options);
+  }
+
+  private async startProcess(options: EventProcessorOptions) {
+    const lastTimestampFunc = options.getLastProcessedTimestamp;
+    if (!lastTimestampFunc) {
+      throw new Error(`Undefined function for getting the last processed timestamp`);
+    }
+
+    const maxHeight = await lastTimestampFunc();
+    if (maxHeight === undefined) {
+      throw new Error(`Cannot get the last processed timestamp via the provided getLastProcessedTimestamp()`);
+    }
+
+    await this.callElasticsearchEvents(maxHeight);
+  }
+
+  async callElasticsearchEvents(lastProcessedTimestamp: number): Promise<void> {
+    const url = `${this.options.elasticUrl}/events/_search?scroll=${this.options.scrollTimeout}`;
```

Comment:
add try/catch block in case of any errors when trying to fetch data from elastic

Reply:
right. added